README.md

```diff
@@ -1,196 +1,289 @@
+          - /opt/seq
+        dx_user_token: <token>
+    mode: debug
+    upload_project: project-id
+  roles:
+    -  dx-streaming-upload
+```
+Here are the instructions for token generation.  
+Launch the ansible-playbook  
+```
+sudo ansible-playbook dx-streaming-upload/dx-upload-play.yml
+```
+Give the right permission to cron
+```
+sudo cron
+```
+#### Using RedHat (tested on 7.4 and 7.5)
+Create a working directory. Select the /opt folder as working directory (in our case we are using use ~/dx)
+```
+mkdir ~/dx 
+cd ~/dx
+```
+Install prerequisites
+```
+sudo yum install git -y
+sudo yum install wget -y
+```
+Enable EPEL Repository for RH 7.*
+```
+wget http://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm 
+sudo rpm -ivh epel-release-latest-7.noarch.rpm 
+Install pip and some essential packages
+sudo yum install python-pip -y 
+sudo cp /usr/bin/python2.7 /usr/bin/python
+sudo pip install -U setuptools 
+sudo pip install packaging
+make
+sudo yum install gcc gcc-c++ kernel-devel -y
+```
+Install ansible
+```
+git clone https://github.com/ansible/ansible.git
+cd ansible/
+make
+sudo make install
+```
+Download or move some test sequencing data in `/opt/seq` folder 
+
+Clone streaming repository
+```
+cd ~/dx
+git clone https://github.com/dnanexus-rnd/dx-streaming-upload.git
+```
+Create dx-upload-play.yml file inside the dx-streaming-folder. Example given [here](#dx-upload-play.yml)  
+Here are the [instructions](https://wiki.dnanexus.com/Command-Line-Client/Login-and-Logout#Generating-an-authentication-token) for token generation.  
+Launch the ansible-playbook 
+```
+sudo ansible-playbook dx-streaming-upload/dx-upload-play.yml 
+```
+## Examples
+#### Role Variables
+- `mode`: `{deploy, debug}` In the *debug* mode, monitoring cron job is triggered every minute; in *deploy mode*, monitoring cron job is triggered every hour.
+- `upload_project`: ID of the DNAnexus project that the RUN folders should be uploaded to. The ID is of the form `project-BpyQyjj0Y7V0Gbg7g52Pqf8q`
+- `dx_token`: API token for the DNAnexus user to be used for data upload. The API token should give minimally UPLOAD access to the `{{ upload project }}`, or CONTRIBUTE access if `downstream_applet` is specified. Instructions for generating a API token can be found at [DNAnexus wiki](https://wiki.dnanexus.com/UI/API-Tokens). This value is overriden by `dx_user_token` in `monitored_users`.
+- `monitored_users`: This is a list of objects, each representing a local user, with its set of incremental upload parameters. For each `monitored_user`, the following values are accepted
+  - `username`: (Required) username of the local user
+  - `monitored_directories`: (Required)  Path to the local directory that should be monitored for RUN folders. Multiple directories can be listed. Suppose that the folder `20160101_M000001_0001_000000000-ABCDE` is the RUN directory, then the folder structure assumed is `{{monitored_dir}}/20160101_M000001_0001_000000000-ABCDE`

```

Comment:
--> "Suppose that the folder `20160101_M000001_0001_000000000-ABCDE` is the RUN directory. Then, the folder structure assumed is `{{monitored_dir}}/20160101_M000001_0001_000000000-ABCDE`"

Reply:
Also: put period at end of all of these bullet points (keep it consistent)